x-pack/plugins/cases/server/authorization/audit_logger.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { OperationDetails } from '.';
+import { AuditLogger, EventCategory, EventOutcome } from '../../../security/server';
+
+enum AuthorizationResult {
+  Unauthorized = 'Unauthorized',
+  Authorized = 'Authorized',
+}
+
+export class AuthorizationAuditLogger {
+  private readonly auditLogger?: AuditLogger;
+
+  constructor(logger: AuditLogger | undefined) {
+    this.auditLogger = logger;
+  }
+
+  private createMessage({
+    result,
+    className,
+    operation,
+  }: {
+    result: AuthorizationResult;
+    className?: string;
+    operation: OperationDetails;
+  }): string {
+    // class: determine what we're calling this (don't use class)
+    const classMsg = className === undefined ? 'of any class' : `of class "${className}"`;
```

Comment:
nit: I usually prefer to do `className == null` because this syntax catches both `undefined` and `null` values.